src/Servers/Kestrel/Core/src/Internal/HttpConnection.cs

```diff
@@ -18,7 +18,8 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
 {
     internal class HttpConnection : ITimeoutHandler
     {
-        private static readonly ReadOnlyMemory<byte> Http2Id = new[] { (byte)'h', (byte)'2' };
+        // Use C#7.3's ReadOnlySpan<byte> optimization for static data https://vcsjones.com/2019/02/01/csharp-readonly-span-bytes-static/
+        private static ReadOnlySpan<byte> Http2Id => new[] { (byte)'h', (byte)'2' };
```

Comment:
Did you intend to remove readonly?

Reply:
It’s a property

Reply:
> // Use C#7.3's ReadOnlySpan<byte> optimization for static data https://vcsjones.com/2019/02/01/csharp-readonly-span-bytes-static/

So you're removing a memcopy, but adding an byte array allocation per call to Http2Id? Or does the allocation get elided? Do we have a benchmark showing this is better?

Reply:
It’s static in the binary not allocated, it’s also done all over kestrel. I haven’t measured but in the grand scheme of things it’s free-er than what’s there right now

Reply:
I think I remember mass changing these now. Back then I think we had a benchmark or dotMemory tracing or something. Either way, I looked at the IL and agree that it's free-er than what we have now.